src/lib/y2firstboot/clients/root.rb

```diff
@@ -70,7 +70,9 @@ def force?
       #
       # @return [Boolean]
       def root_password_from_user?
-        Y2Firstboot::Clients::User.user_password == Y2Firstboot::Clients::User.root_password
+        user_password = Y2Firstboot::Clients::User.user_password
+
+        user_password && user_password == Y2Firstboot::Clients::User.root_password
```

Comment:
well, this kind of breaks contract and documentation of method as it can return nil.
so I would use something like
```suggestion
        return false unless user_password

        user_password == Y2Firstboot::Clients::User.root_password
```

Reply:
You're right. In fact, I wrote the code similarly initially (_return false unless Y2Firstboot::Clients::User.user_password_) but then I discarded it for using the version commited. But it's true, it can return nil which is not desired.

Suggestion committed, thanks!